app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/AvatarsDownloader.kt

```diff
@@ -47,20 +49,37 @@ class AvatarsDownloader(
     /** download avatar for the given user and a known avatar url */
     suspend fun download(userId: Long, avatarUrl: String) {
         if (!ensureCacheDirExists()) return
-        val avatarFile = File(cacheDir, "$userId")
+        val avatarPath = avatarPath(userId)
         try {
             val response = httpClient.get(avatarUrl) {
                 expectSuccess = true
             }
-            response.bodyAsChannel().copyAndClose(avatarFile.writeChannel())
-            Log.d(TAG, "Downloaded file: ${avatarFile.path}")
+            filesystem.write(avatarPath) {
+                response.bodyAsChannel().copyTo(this.outputStream())
```

Comment:
`outputStream` and `copyTo` is a Java API.

`this` is a `okio.BufferedSink` but the response provides a `ByteReadChannel`. It is possible to write a `ByteReadChannel::copyTo(out: okio.BufferedSink)` oneself in case no adapter library for that exists, it is indeed not that much. However, compared with photo uploads, OSM user avatars are quite small, so in this case it would be 100% fine to just convert to ByteArray and stuff that ByteArray into the okio sink. (But maybe not that consistent if conversion in one direction is implemented but not the other - optimum anyway would be if there was an official library for that)

Soem details in the other comment.